dotnet-test-samples/kinesis-lambda-dynamodb/tests/KinesisEventHandler.UnitTests/Handlers/KinesisEventHandlerTests.cs

```diff
@@ -0,0 +1,90 @@
+using System.Threading.Tasks;
+using Amazon.Lambda.Core;
+using Amazon.Lambda.TestUtilities;
+using FluentAssertions;
+using KinesisEventHandler.Handlers;
+using KinesisEventHandler.Models;
+using KinesisEventHandler.UnitTests.Utilities;
+using Moq;
+using Xunit;
+
+namespace KinesisEventHandler.UnitTests.Handlers;
+
+public class KinesisEventHandlerTests
+{
+    private readonly Mock<KinesisEventHandler<Employee>> _mockKinesisEventTrigger;
+
+    public KinesisEventHandlerTests()
+    {
+        _mockKinesisEventTrigger = new Mock<KinesisEventHandler<Employee>>();
```

Comment:
Usually it's best to avoid initializing mocks in the c'tor/setup as future tests may add more functionality effecting other tests without realizing it.

Reply:
That's right but maybe instead try extracting the initialize logic which you do not care about in the test to a method or a builder that way although you do need to call it the overhead is small (search online for DAMP vs. DRY) and if done right - i.e. not general names call the explicit call makes sure that no one accidentally ruin your test in the next iteration